phpseclib/Math/BigInteger.php

```diff
@@ -752,18 +750,17 @@ function __toString()
     /**
      * __clone() magic method
      *
-     * Although you can call BigInteger::__toString() directly in PHP5, you cannot call BigInteger::__clone() directly
-     * in PHP5.  You can in PHP4 since it's not a magic method, but in PHP5, you have to call it by using the PHP5
-     * only syntax of $y = clone $x.  As such, if you're trying to write an application that works on both PHP4 and
-     * PHP5, call BigInteger::copy(), instead.
-     *
      * @access public
-     * @see self::copy()
      * @return \phpseclib\Math\BigInteger
      */
     function __clone()
     {
-        return $this->copy();
+        $temp = new static();
+        $temp->value = $this->value;
+        $temp->is_negative = $this->is_negative;
+        $temp->precision = $this->precision;
+        $temp->bitmask = $this->bitmask;
+        return $temp;
     }
```

Comment:
Since this is PHP >= 5 code (because type hinting was added below), could it be that the whole `__clone()` method can be removed? (It looks like it's only needed for modifying **internal objects** after cloning, not **internal variables**. And `value`, `is_negative`, `precision` and `bitmask` are not objects - their values should be copied automatically.) See, e.g., https://stackoverflow.com/questions/2144506/what-is-object-cloning-in-php.